[ADD] l10n_ph

tong-odoo opened the pull request:
- add philippines localization

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/81280).

fja-odoo commented on addons/l10n_ph/README.md:
```diff
@@ -0,0 +1,7 @@
+# Ps-Tech module : l10n_ph
```
No need for the readme here

fja-odoo commented on addons/l10n_ph/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "Philippines - Accounting",
+    "summary": """
+        This is the module to manage the accounting chart for Philippines.
```
The Philippines

fja-odoo commented on addons/l10n_ph/models/account_tax_template.py:
```diff
@@ -0,0 +1,17 @@
+class AccountTaxTemplate(models.Model):
+    _inherit = "account.tax.template"
+
+    atc = fields.Char("Philippines ATC")
```
l10n_ph_atc will give more info on the field

tong-odoo replied:
help info?

fja-odoo replied:
I mean naming the field `l10n_ph_atc` will be better so that people know this field is part of this localization

fja-odoo commented on addons/l10n_ph/models/account_tax.py:
```diff
@@ -0,0 +1,8 @@
+class AccountTax(models.Model):
+    _inherit = "account.tax"
+
+    l10n_ph_description = fields.Char("Philippines Description")
```
What is the point of this field ?
There is already a description field: https://github.com/odoo/odoo/blob/14.0/addons/account/models/chart_template.py#L917
Why make a copy ?

tong-odoo replied:
I asked JGR and she said this is required.

fja-odoo replied:
But below in `_get_tax_vals` you use the field `description` to fill `l10n_ph_description` so it is just a duplicate of the original description isn't it ?

qdp-odoo commented on addons/l10n_ph/data/account_fiscal_position_template.xml:
```diff
@@ -0,0 +1,32 @@
+        <field name="name">Zero-Rated</field>
+        <field name="chart_template_id" ref="l10n_ph_chart_template" />
+        <field name="auto_apply" eval="True" />
+        <field name="country_id" ref="base.ph" />
```
I don't see any difference between non-vat, vat exempt, and zero-rated fiscal positions conditions of application, which means that the last 2 won't be used. Do I miss something? why are they there?

Also the zero-rated, vat exempt, and non-vat taxes have the exact same effect on journal entries (and there is no vat report), so again these seem useless?

tong-odoo replied:
Its from the government, we still need to include one of them, we will include vat exempt only.

qdp-odoo commented on addons/l10n_ph/data/account_fiscal_position_tax_template.xml:
```diff
@@ -0,0 +1,160 @@
+        <field name="tax_dest_id" ref="tax_template_purchase_zero_rated" />
+        <field name="position_id" ref="fiscal_position_template_zero_rated" />
+    </record>
+</odoo>
```
no purchase taxes in the fiscal positions at all?

tong-odoo replied:
Yes we don't need it because the all our clients will edit them manually in the database. There is no standard fiscal position taxes in ph. For the vat_exempt, they will know that from the beginning the clients are vat_exempt or not.

qdp-odoo commented on addons/l10n_ph/data/account_tax_template_cash.xml:
```diff
@@ -0,0 +1,2803 @@
+<?xml version='1.0' encoding='utf-8' ?>
```
what's that file for? 

is that all duplicated taxes meant to be cash-based (but with cash basis unset for any reason?) ??

qdp-odoo commented on addons/l10n_ph/models/account_tax.py:
```diff
@@ -0,0 +1,7 @@
+class AccountTax(models.Model):
+    _inherit = "account.tax"
+
+    l10n_ph_atc = fields.Char("Philippines ATC")
```
what's ATC? can you add a help tooltip ?


I don't see any use of this field anywhere, btw.

tong-odoo replied:
Its for the BIR report. Its init from the Template so we can use it in the BIR report in later module

qdp-odoo reviewed (changes requested):
I've some remarks.

Also, 160 purchase taxes... dude, isn't that a bit overkill?

qdp-odoo commented:
to answer your question about the number of taxes: we don't want to manage (by default) all rare cases of taxes. So, for example, if there are special cases of taxes that are used for alcohol, drugs.. (excises), we don't want them because it's very unlikely that a regular company would need it. By a matter of fact, those taxes are often not VAT and have their own report (that we don't manage by default as well)

So, based on that lone and somewhat abstract sentence ^^, do you think a lambda company from the Philipines could use any of the 160 purchase taxes you defined?

tong-odoo commented:
Thanks I will discuss this with FC first. We will try to trim down a bit and come back to you

qdp-odoo commented on addons/l10n_ph/data/ir_actions_server.xml:
```diff
@@ -0,0 +1,36 @@
+    pass
+else:
+    raise UserError('Only Outbound Payment is available.')
+]]></field>
```
??? what does it do

qdp-odoo commented on addons/l10n_ph/models/res_partner.py:
```diff
@@ -0,0 +1,30 @@
+                raise UserError(_("Tax ID for PH contacts must be 17 characters long! (NNN-NNN-NNN-NNNNN)"))
+            r = re.compile(r"\d{3}-\d{3}-\d{3}-\d{5}")
+            if r.match(record.vat) is None:
+                raise UserError(_("Tax ID for PH contacts must follow the following numeric format: NNN-NNN-NNN-NNNNN"))
```
I don't like that it's not in base_vat. Any reason why it's not there?

tong-odoo commented on addons/l10n_ph/controllers/main.py:
```diff
@@ -0,0 +1,83 @@
+
+
+class BirFormController(http.Controller):
+    @http.route('/bir_2307/download/xls/<move_ids>', type="http", auth="user", methods=["GET"])
```
Bcs i don't want to create an attachment to store the attachment in the system. Or which one is the standard way? Is there any file I can reference to?

tong-odoo commented on addons/l10n_ph/controllers/main.py:
```diff
@@ -0,0 +1,83 @@
+                    values['atc'] = tax.l10n_ph_atc
+                    values['price_subtotal'] = invoice_line.price_subtotal
+                    values['amount'] = tax.amount
+                    values['tax_amount'] = tax._compute_amount(invoice_line.price_subtotal, invoice_line.price_unit)
```
Is there other way to affect the tax if I don't compute the tax from tax details? Sorry I don't have much knowledge on tax computation.

fja-odoo commented:
@qdp-odoo Hello Quentin, friendly reminder for this PR :)

FlorianGilbert commented on addons/l10n_ph/wizard/generate_2307_wizard_views.xml:
```diff
@@ -0,0 +1,36 @@
+                            <field name="currency_id" invisible="1"/>
+                            <field name="amount_tax_signed" string="Tax" sum="Total" optional="hide" modifiers="{'readonly':true}" widget="monetary" options="{'currency_field': 'currency_id'}"/>
+                            <field name="amount_total_signed" string="Total" sum="Total" decoration-bf="1" optional="show" widget="monetary" options="{'currency_field': 'currency_id'}"/>
+                            <field name="state" widget="badge" decoration-success="state == 'posted'" decoration-info="state == 'draft'" optional="show" on_change="1" modifiers="{'readonly':true,'required':true}"/>
```
```suggestion
                            <field name="state" widget="badge" decoration-success="state == 'posted'" decoration-info="state == 'draft'" optional="show" on_change="1" modifiers="{'readonly':true, 'required':true}"/>
```

FlorianGilbert commented on addons/l10n_ph/wizard/generate_2307_wizard_views.xml:
```diff
@@ -0,0 +1,36 @@
+                    </field>
+                </group>
+                <footer>
+                    <button string="Generate" type="object" name="action_generate" class="btn-primary" data-hotkey="q"/>
+                    <button string="Cancel" special="cancel" data-hotkey="z" class="btn-secondary"/>
```
```suggestion
                    <button string="Generate" type="object" name="action_generate" class="btn btn-primary" data-hotkey="q"/>
                    <button string="Cancel" special="cancel" data-hotkey="z" class="btn btn-secondary"/>
```
https://getbootstrap.com/docs/5.2/components/buttons/

tong-odoo commented on addons/l10n_ph/data/account.account.template.csv:
```diff
@@ -0,0 +1,99 @@
+"a110201","Input VAT 12%","110201","account.data_account_type_current_assets","l10n_ph_chart_template","False"
+"a110202","Deferred Input VAT 12%","110202","account.data_account_type_current_assets","l10n_ph_chart_template","False"
+"a110203","Deferred Tax Asset","110203","account.data_account_type_current_assets","l10n_ph_chart_template","False"
+"a110204","Def VAT -> 1M","110204","account.data_account_type_current_assets","l10n_ph_chart_template","True"
```
I am asking FC for this one

tong-odoo replied:
Its for if the deferred value added tax (VAT) amount is 1Million. It is not commonly used among the client so we remove this account from the template.

FlorianGilbert reviewed (changes requested):
Sorry for the late review, probably the last one :)

tong-odoo commented:
@FlorianGilbert Thanks for the review. I updated the PR based on your comment. 

However, since I am working on MY localisation as well and notice that we need to update the account.account.template id to `l10n_my_11`. Please kindly let me know if I need to do the same for Philippines. Thanks!

FlorianGilbert commented:
> @FlorianGilbert Thanks for the review. I updated the PR based on your comment.
> 
> However, since I am working on MY localisation as well and notice that we need to update the account.account.template id to `l10n_my_11`. Please kindly let me know if I need to do the same for Philippines. Thanks!

Yes please and it will be good to be merged! :)

tong-odoo commented:
> > @FlorianGilbert Thanks for the review. I updated the PR based on your comment.
> > However, since I am working on MY localisation as well and notice that we need to update the account.account.template id to `l10n_my_11`. Please kindly let me know if I need to do the same for Philippines. Thanks!
> 
> Yes please and it will be good to be merged! :)

Updated. Thanks for you help again!

FlorianGilbert reviewed (approved):
@robodoo r+

robodoo commented:
@tong-odoo @FlorianGilbert because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

FlorianGilbert commented:
@robodoo rebase-ff

robodoo commented:
Merge method set to rebase and fast-forward.

h4818 commented:
@FlorianGilbert is there a reason we don't have a demo company for this l10n?
